tests/test_jointcal.py

```diff
@@ -174,6 +174,51 @@ def test_iterateFit_exceedMaxSteps(self):
         self.assertEqual(self.fitter.minimize.call_count, maxSteps)
         log.error.assert_called_with("testing failed to converge after %s steps" % maxSteps)
 
+    def test_moderate_chi2_increase(self):
+        """DM-25159: warn, but don't fail, on moderate chi2 increases between
+        steps.
+        """
+        chi2_1 = lsst.jointcal.chi2.Chi2Statistic()
+        chi2_1.chi2 = 100.0
+        chi2_1.ndof = 100
+        chi2_2 = lsst.jointcal.chi2.Chi2Statistic()
+        chi2_2.chi2 = 300.0
+        chi2_2.ndof = 100
+
+        chi2s = [self.goodChi2, chi2_1, chi2_2, self.goodChi2, self.goodChi2]
+        self.fitter.computeChi2.side_effect = chi2s
+        self.fitter.minimize.side_effect = [MinimizeResult.Chi2Increased,
+                                            MinimizeResult.Chi2Increased,
+                                            MinimizeResult.Chi2Increased,
+                                            MinimizeResult.Converged,
+                                            MinimizeResult.Converged]
+        with lsst.log.UsePythonLogging():  # so that assertLogs works with lsst.log
+            with self.assertLogs("jointcal", level="WARN") as logger:
+                self.jointcal._iterate_fit(self.associations, self.fitter,
+                                           self.maxSteps, self.name, self.whatToFit)
+            msg = "WARNING:jointcal:Significant chi2 increase by a factor of 300 / 100 = 3"
+            self.assertIn(msg, logger.output)
```

Comment:
Indentation looks like it is off by one here; `logger` is not defined at this level.

Reply:
Context manager variables remain outside the block (`with` statements do not create new scope), and it's often necessary to access them outside, e.g. for `assertRaises`. See here:

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs

Reply:
Wow, ok!